projects/ccf-eui/src/app/shared/components/spatial-search-keyboard-ui-behavior/spatial-search-keyboard-ui-behavior.component.ts

```diff
@@ -0,0 +1,77 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, HostListener, Input, Output } from '@angular/core';
+
+export interface SpatialSearchPosition {
+  x: number;
+  y: number;
+  z: number;
+}
+
+@Component({
+  selector: 'ccf-spatial-search-keyboard-ui-behavior',
+  templateUrl: './spatial-search-keyboard-ui-behavior.component.html',
+  styleUrls: ['./spatial-search-keyboard-ui-behavior.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SpatialSearchKeyboardUIBehaviorComponent {
+  @HostBinding('class') readonly className = 'ccf-spatial-search-keyboard-ui-behavior';
+
+  @Input() delta = 1;
+
+  @Input() shiftDelta: number;
+
+  @Input() position: SpatialSearchPosition = {
+    x: 0,
+    y: 0,
+    z: 0
+  };
+
+  @Output() readonly changePosition = new EventEmitter<SpatialSearchPosition>();
+
+  currentDirection?: string;
+
+  currentKey?: string;
+
+  @HostListener('document:keydown', ['$event'])
```

Comment:
You'll need to detect the shift key here and change the delta (and show in UI) appropriately.